aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -302,13 +303,51 @@ impl AptosVMImpl {
             .or_else(|err| convert_prologue_error(chain_specific_info, err, log_context))
     }
 
+    fn verify_module_bundle<S: MoveResolverExt>(
+        session: &mut SessionExt<S>,
+        module_bundle: &ModuleBundle,
+        txn_data: &TransactionMetadata,
+    ) -> VMResult<()> {
+        for module_blob in module_bundle.iter() {
+            match CompiledModule::deserialize(module_blob.code()) {
+                Ok(module) => {
+                    // verify the sender matches the address
+                    if module.address() != &txn_data.sender {
+                        return Err(verification_error(
+                            StatusCode::MODULE_ADDRESS_DOES_NOT_MATCH_SENDER,
+                            IndexKind::AddressIdentifier,
+                            module.self_handle_idx().0,
+                        )
+                        .finish(Location::Undefined));
+                    }
```

Comment:
this is also checked upstream, is there a point in checking it again?

Reply:
this is validation, so we can fail before execution